prow/plugins/golint/suggestion/golint_suggestion.go

```diff
@@ -90,6 +92,23 @@ func fixStutter(p lint.Problem) string {
 	return suggestion
 }
 
+func fixErrorf(p lint.Problem) string {
+	matches := lintErrorfRegex.FindStringSubmatch(p.Text)
+	if len(matches) != 3 {
+		return ""
+	}
+	reErrorMessage := regexp.MustCompile(`errors\.New\(fmt\.Sprintf(.*)\)`)
```

Comment:
I don't think it is possible to make a golang regexp that can match the correct parentheses in all cases. (Consider parentheses inside the `Sprintf` and parentheses surrounding the entire `errors.New` statement.) This is a limitation of FSA based regular expression engines: https://stackoverflow.com/a/46334384

I'm not sure what the cleanest way to handle this is.

Reply:
I see. I just updated the logic to extract the error message myself. Do you have any suggestions on how I can simplify this?